samples/sensor/magn_polling/src/main.c

```diff
@@ -10,10 +10,16 @@
 #include <zephyr/sys/printk.h>
 #include <stdio.h>
 
-static void do_main(const struct device *dev)
+void main(void)
 {
-	int ret;
+	struct device *dev = DEVICE_DT_GET_ONE(bmc150_magn);
```

Comment:
note: this sample is not generic anyway, we should likely rely on aliases, otherwise it's specific to `bmc150_mgn`, which btw should be `bosch_bmc150_magn`. does it compile now?

Reply:
Should we come up with an alias now or is that for later?
Actually it didn't compile... What's the best way to compiling a sample? Using west build I guess I'm stuck to either finding a board with the sensor in the DTS already or just adding a temporary app.overlay in the sample directory?

Reply:
You can use `twister`, this way you'll make use of the information in `sample.yaml`. I've always found weird that `west` doesn't make use of `sample.yaml` to tell you _hey, the board you're building for is not compatible!_, we can probably improve that for a better user experience.

Reply:
Thanks, I should definitely take a closer look at twister.